lobster/tools/python/python.py

```diff
@@ -90,7 +90,11 @@ def fqn(self):
             rv = self.parent.fqn() + "."
         else:
             rv = ""
-        rv += self.name
+        if self.location.line is not None and \
```

Comment:
Please consider the following refactoring:

1. Call `self.fqn` once before creating `Implementation`, see https://github.com/bmw-software-engineering/lobster/blob/86cd606a0d624e663c57fa298b88a11f5f2ed4a8/lobster/tools/python/python.py#L157
2. Find a unique name by iterating over `items`
3. Don't use the line number, because this ID will be dynamic and changes if someone adds new lines to the file. Instead, use some kind of counter.

The first tag name shall always be without a number. Only the second tag name shall add a suffix.